### ISSUE TITLE
NEW_RELIC_AGENT_ENABLED does not disable agent

### ISSUE THREAD

**Issue by mehyedes** [newrelic/newrelic-ruby-agent#339]
We have been trying to disable the ruby agent for some of our rake processes in order to debug some production issues. But seems like setting `NEW_RELIC_AGENT_ENABLED=false` is not doing anything.
This is how we are trying to disable the agent:
```
NEW_RELIC_AGENT_ENABLED=false bundle exec rake ...
```
However, we can still see that there is traffic going to NR servers for these rake processes:
```
# Using lsof
bundle    25355     deploy   29u  IPv4 80303490      0t0  TCP <hostname>:47884->162.247.243.149:https (ESTABLISHED)
```
In the agent's configuration file, the config is set to `auto`:
```
  agent_enabled: auto
```

This is the gem version that we are using:
```
gem 'newrelic_rpm', '~> 6.5.0'
```
and our ruby version is `2.6.5`

Shouldn't the environment variable take precedence over the agent's configuration file?
Or maybe we are doing it wrong?

**Comment by mehyedes** [comment]
Ok, I have been reading the [docs](https://docs.newrelic.com/docs/agents/ruby-agent/troubleshooting/control-when-ruby-agent-starts), and it seems that setting `NEW_RELIC_AGENT_ENABLED=false` is not enough.
`NEW_RELIC_MONITOR_MODE` must also be set to `false` to make sure the agent does not start or transmit any data.
And now I can see this log message in the new relic logs:
```
Agent configured not to send data in this environment.
```
I think the relation between `monitor_mode` and `agent_enables` should be further explained in the docs, as it was only possible to understand this behaviour by trial and error.

**Comment by rachelleahklein** [comment]
Thanks so much for bringing this to our attention. We'll make sure the documentation on this is clear.

**Comment by angelatan2** [comment]
This was closed by the stale bot inadvertently.  Our apologies, we would like to review this request for documentation clarification.

**Comment by amhuntsman** [comment]
@mehyedes Apologies for the long response time on this issue. It's difficult to review the documentation from June 2020, but I've reviewed the current docs and the code for `newrelic_rpm` version 6.5.0, and to the best of my understanding the intent is this:

Setting either `agent_enabled` or `monitor_mode` to `false`, via `newrelic.yml` or environment variables, should preclude any instrumentation data from being sent to the collector.

I tested the gem using a sample Rails application with `agent_enabled: auto` in `newrelic.yml`, ran a rake task designed to generate network traffic, first setting NEW_RELIC_AGENT_ENABLED=false, then NEW_RELIC_MONITOR_MODE=false, then with no environment variables passed. In the first two cases, no data was seen to be sent to New Relic; in the last case, request data was shown in the NR frontend.

Barring more detailed network monitoring data, I believe that these options are working as intended.
